cetmix_tower_server/models/cx_tower_variable_value.py

```diff
@@ -277,3 +284,29 @@ def _check_single_assignment(self):
                         "Server, Server Template, or Plan Line Action."
                     )
                 )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # Extract all unique refs from vals_list
+        variable_refs = self._prepare_references(
+            "cx.tower.variable", "variable_id", vals_list
+        )
+        # Initialize line indices for each variable_id
+        variable_line_index_dict = defaultdict(int)
+
+        # Populate vals with references
+        for vals in vals_list:
+            variable_id = vals.get("variable_id")
+            if variable_id:
+                variable_line_index_dict[variable_id] += 1
+                line_index = variable_line_index_dict[variable_id]
+
+                vals["reference"] = f"{variable_refs[variable_id]}_{line_index}"
+            else:
+                # Handle the case where variable_id is not present
+                variable_line_index_dict["no_variable"] += 1
+                line_index = variable_line_index_dict["no_variable"]
+                vals["reference"] = f"value_with_no_variable_{line_index}"
+
+        # Delegate creation to the superclass method
+        return super().create(vals_list)
```

Comment:
this method has the same code and is used in three different places, please make one common method (maybe in cx.tower.reference.mixin model) and use it, it will make it easier to maintain the code in the future

Reply:
Thanks, I'll fix it